applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h

```diff
@@ -193,6 +205,61 @@ class GeometryUtilities
     	KRATOS_CATCH("");
     }
 
+    // --------------------------------------------------------------------------
+    void FlagNodesInRadius( ModelPart::NodesContainerType& rNodeSet, const Variable<bool>& rFlagVariable, double filter_radius)
+    {
+        unsigned int bucket_size = 100;
+
+        // Create search tree
+        if(mIsSearchTreeConstructed==false)
```

Comment:
I have the feeling that creating a node search as a member of the very general GeometryUtilities Class just in order to save some time when calling this function several times is a little over the top. 
The class contains many independent functionalities and the node search is only needed in this function.

Recreating the search three each time this function is called would make more sense to me.